### PR TITLE
Add Range cover? vs include?

### PR DESCRIPTION
**Pull request by gylaz**

### CODE REVIEW

**Comment by timoschilling** [comment]
The reason is that `includes?` converts the range to an array, `covers?` works without that step.

File: ruby/range-cover-vs-include.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Range cover? vs include? method
+
+The `cover?` method yields better performance than `include?`,
+however there are subtle differences to when one should be used over other.
```
Comment (JoelQ) [inline ruby/range-cover-vs-include.md]: Can we quantify these performance improvements?

Comment (gylaz) [inline ruby/range-cover-vs-include.md]: What are you thinking? Providing the benchmark for the users to run, or just saying something like `~ X times faster`?

Comment (JoelQ) [inline ruby/range-cover-vs-include.md]: Either would work. I just think we need something a bit more specific than a vague "it's faster"

Comment (zamith) [inline ruby/range-cover-vs-include.md]: What about both? "It's X times faster when running this benchmark".

Comment (thorncp) [inline ruby/range-cover-vs-include.md]: Running a simple benchmark with [readygo] and including the output could be interesting.

[readygo]: https://github.com/garybernhardt/readygo

Comment (rShetty) [inline ruby/range-cover-vs-include.md]: I found one of the benchmarks for these here 
https://gist.github.com/inkel/6090991

Results:
                             user     system      total        real
#include? - best case    0.130000   0.000000   0.130000 (  0.129843)
#include? - worst case   0.130000   0.000000   0.130000 (  0.130545)
#include? - avg. case    0.130000   0.000000   0.130000 (  0.127180)
#cover? - best case      0.130000   0.000000   0.130000 (  0.127623)
#cover? - worst case     0.130000   0.000000   0.130000 (  0.127207)
#cover? - avg. case      0.120000   0.000000   0.120000 (  0.127252)

**Comment by stefannibrasil** [comment]
This repo is no longer active and it's being archived. The contents will still be available as read-only. Thank you for your contributions!

**Pull request closed without merging**